openhcl/underhill_core/src/dma_manager.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::{Arc, Mutex, Weak};
+use memory_range::MemoryRange;
+use once_cell::sync::OnceCell;
+
+pub use dma_client::{DmaClient, DmaInterface, DmaTransaction, DmaTransactionHandler};
+
+pub enum DmaError {
+    InitializationFailed,
+    MapFailed,
+    UnmapFailed,
+    PinFailed,
+    BounceBufferFailed,
+}
+
+static GLOBAL_DMA_MANAGER: OnceCell<Arc<GlobalDmaManager>> = OnceCell::new();
+
+/// Global DMA Manager to handle resources and manage clients
+pub struct GlobalDmaManager {
```

Comment:
What settings will this manager have? Which of those do you expect to expose in Vtl2Settings?